hera_cal/frf.py

```diff
@@ -616,10 +956,29 @@ def tophat_frfilter(self, keys=None, wgts=None, mode='clean',
          see uvtools.dspec.dayenu_filter for key formats.
         read_cache: bool, If true, read existing cache files in cache_dir before running.
         write_cache: bool. If true, create new cache file with precomputed matrices
-         that were not in previously loaded cache files.
-        cache: dictionary containing pre-computed filter products.
-        skip_flagged_edges : bool, if true do not include edge times in filtering region (filter over sub-region).
+            that were not in previously loaded cache files.
+        center_before_filtering: bool, optional
+            shift the data by multiplying by the center fringe-rate and filter a window centered at zero fringe rate.
+            This improves filter stability when the filtering window is highly offset from zero since we avoid interpolating
+            with fine-time-scale modes.
+        fr_freq_skip: int, optional
+            bin fringe rates from every freq_skip channels.
+            default is 1 -> takes a long time. We recommend setting this to be larger.
         verbose: bool, optional, lots of outputs!
+        dfr: float, optional.
+            spacing of fringe-rate grid used to perform binning and percentile calc
+            in units of mHz.
+            default is None -> set to 1 / (dtime * Ntime) of uvd.
+        nfr: float, optional.
+            number of points on fringe-rate grid to perform binning and percentile calc.
+            default is None -> set to uvd.Ntimes.
+        pre_filter_modes_between_lobe_minimum_and_zero: bool, optional
```

Comment:
if this is the case, isn't the answer to just change the FRP such that it has zero power near frate ~ 0?

Reply:
This can be done if we are using the DFT basis to perform our fringe-rate filter. However, it will not work if we are using functions that are not localized in fringe-rate space including both DPSS modes and DFT modes with side-lobes induced by missing data and finite bandpass.